docs/content/mongocxx-v3/policies/abi-versioning.md

```diff
@@ -0,0 +1,110 @@
++++
+date = "2024-02-26T00:00:00+00:00"
+title = "ABI Versioning"
+[menu.main]
+  weight = 33
+  parent="mongocxx3/policies"
++++
+
+The CXX Driver uses Ulrich Drepper's guide to "Defining Stability", the Linux shared library "soname" convention, and the CMake `SOVERSION` target property for ABI versioning.
+
+Per [Ulrich Drepper](https://www.cs.dartmouth.edu/~sergey/cs258/ABI/UlrichDrepper-How-To-Write-Shared-Libraries.pdf):
+
+> The definition of stability should therefore use the _documented_ interface as the basis. Legitimate uses of interfaces should not be affected by changes in the implementation; using interfaces in an undefined way void the warranty. The same is true for using completely undocumented, internal symbols. Those must not be used at all.
+
+Per [The Linux Documentation Project](https://tldp.org/HOWTO/Program-Library-HOWTO/shared-libraries.html):
+
+> Every shared library has a special name called the \`\`soname''. The soname has the prefix \`\`lib'', the name of the library, the phrase \`\`.so'', followed by a period and a version number that is incremented whenever the interface changes (as a special exception, the lowest-level C libraries don't start with \`\`lib''). A fully-qualified soname includes as a prefix the directory it's in; on a working system a fully-qualified soname is simply a symbolic link to the shared library's \`\`real name''.
+
+Per [CMake documentation](https://cmake.org/cmake/help/latest/prop_tgt/SOVERSION.html):
+
+> For shared libraries `VERSION` and `SOVERSION` can be used to specify the build version and ABI version respectively. When building or installing appropriate symlinks are created if the platform supports symlinks and the linker supports so-names. If only one of both is specified the missing is assumed to have the same version number.
+
+For purposes of ABI versioning, we distinguish between the _stable ABI_ and the _unstable ABI_.
+
+The ABI version number is bumped whenever there is a backward incompatible change to the stable ABI as defined below.
+
+**IMPORTANT:** Only the stability of the stable ABI is communicated by the ABI version number. Backward (in)compatible changes to the unstable ABI are not communicated by the ABI version number.
+
+**NOTE:** The ABI versioning policy is only applicable to _shared libraries_. It does not apply to static libraries.
+
+## Binary Compatibility
+
+An _ABI namespace_ is declared within the library's root namespace (see [Source Compatibility]({{< relref "api-versioning#source-compatibility" >}})), prefixed with the letter `v` and followed by either an integer or `_noabi`.
```

Comment:
Where is this supposed to link to?

Reply:
It links to the new "Source Compatibility" header in `api-versioning.md`: https://github.com/mongodb/mongo-cxx-driver/pull/1102/files#diff-c5a6c43199b7f312d661e08521043dc28fc1341a58ade10b1402a74fcb0453c0R26